src/cpp/src/llm_pipeline.cpp

```diff
@@ -251,24 +265,32 @@ class StatefulLLMPipeline final : public LLMPipelineImplBase {
             utils::initialize_position_ids(*position_ids, attention_mask, kv_cache_len);
         }
 
+        if(m_adapter_controller) {
```

Comment:
BTW, since CB implementation can be used not only for serving, but for clients as well, we need to reuse this feature there as well.